include/signal.h

```diff
@@ -386,7 +394,11 @@
  */
 
 #ifndef __SIGSET_T_DEFINED
-typedef uint32_t sigset_t;   /* Bit set of 32 signals */
+struct sigset_s
+{
+  uint32_t _elem[_SIGSET_NELEM];
+};
```

Comment:
Why not `typedef uint32_t sigset_t[2];`? So
```
#define NULL_SIGNAL_SET     { (uint32_t)0x00000000, (uint32_t)0x00000000 }
#define ALL_SIGNAL_SET    { (uinr32_t)0xffffffff, (uinr32_t)0xffffffff }
```
as this would require only to replace conditions like `if (unmaskedset == NULL_SIGNAL_SET)` with `if (nxsig_isequal(unmaskedset, NULL_SIGNAL_SET))`.

Reply:
Normally you would wrap any array type in a structure.  That is done is several places for example in include/sys/select.c in the definition of fd_set type.

All OS code I have looked at does this.  I think it has to do with problems that can arise if an array is used in certain circumstances.  But I would need to investigate that.  Certainly wrapping the array in a struct is more common and guaranteed to be safe in all usage.

Reply:
Yeah. This would cover most of the init assignments, but `oact->sa_mask = NULL_SIGNAL_SET;` will not work.

Reply:
Let's not spend time on this and follow with struct wrapped array.

Reply:
In most interfaces, signal sets are passed as pointers to signal sets.  Like:

    int some_inteface(sigset_t *set);
    sigset_t set;
    ....
    some_interface(&set);

If sigset_t is an array, when an array is passed, it is already passed as a pointer.  So the argument is really like:

    int some_inteface(uint32_t *set[2]);

That is not wrong, but it is really confusing, prone to errors, and generates really mysterious (but correct) warnings.

At one time, I used "char *envp[]" in the prototype of main() in NuttX.  That was a nightmare due to warnings and awkward casts.  I fixed that by changing it to "char **envp" which works pretty much warning free.

Wrapping the array in the structure avoids any potential odd behavior.